components/esp32/panic.c

```diff
@@ -212,7 +212,10 @@ void panicHandler(XtExcFrame *frame)
                     //debugcause if the cause is watchdog 1 and clearing it if it's watchdog 0.
                     if (debugRsn&(1<<8)) {
 #if CONFIG_FREERTOS_WATCHPOINT_END_OF_STACK
-                        panicPutStr("Stack canary watchpoint triggered ");
+                        const char *name = pcTaskGetTaskName(xTaskGetCurrentTaskHandleForCPU(core_id));
```

Comment:
task name provided by the user is copied into [this field](https://github.com/espressif/esp-idf/blob/1e0710f1b24429a316c9c34732aa17bd3f189421/components/freertos/tasks.c#L175)